src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int numberOfFigureTypes = 5;
+        int figureType = random.nextInt(numberOfFigureTypes);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0 :
+                return new Circle(color, random.nextDouble() * 100);
```

Comment:
create `private double getRandomSide()` method

All magic numbers in your code should be constants